tianshou/data/collector.py

```diff
@@ -126,10 +131,20 @@ def reset_buffer(self, keep_statistics: bool = False) -> None:
 
     def reset_env(self) -> None:
         """Reset all of the environments."""
-        obs = self.env.reset()
-        if self.preprocess_fn:
-            obs = self.preprocess_fn(obs=obs,
-                                     env_id=np.arange(self.env_num)).get("obs", obs)
+        if self.gym_reset_return_info:
+            obs, info = self.env.reset(return_info=True)
```

Comment:
I think you cannot do this, at least envpool will fail

Reply:
see my other comment just now https://github.com/thu-ml/tianshou/pull/613#discussion_r886324409 . With my proposal we shouldn't run into this issue. Open to other ideas as well.

Reply:
should be fixed now